dnf/base.py

```diff
@@ -2794,6 +2794,20 @@ def _nevra(item):
 
         return skipped_conflicts, skipped_dependency
 
+    def reboot_needed(self):
+        """Check whether a system reboot is recommended following the transaction
+
+        :return: bool
+        """
+        if not self.transaction:
+            return False
+
+        # List taken from DNF needs-restarting
+        need_reboot = frozenset(('kernel', 'kernel-rt', 'glibc',
+                                'linux-firmware', 'systemd', 'dbus',
+                                'dbus-broker', 'dbus-daemon'))
```

Comment:
It would be nice to share this set with `needs-restarting` plugin somehow.
One option would be a new config option. In this case it should be implemented also for dnf5.

Reply:
Yeah, if we agree that it is useful, it will be spread across libdnf, dnf, dnf-plugins-core, and dnf5 :)
But I'm not gonna block this PR because of that. It could be done in subsequent patches.

Reply:
Yeah I think we can hold off for now and err on the side of minimal changes. Hopefully this list doesn't change too much over the years. But it would be good for it to be configurable by the user.